### PR TITLE
PGFCSB/OSHA update: Various workplace hazards now are hazards

### PR DESCRIPTION
**Pull request by rye-rice**
## About The Pull Request

September 5th, 505 FSC. A Dwayne-class mining vessel- the ISV Crimson Rose- was landed on a sand planet, when it suddenly exploded. The cause  was an untrained engineer  accidentlly unwrenched a fuel pipe filling the area with invisible hydrogen gas. The gas  found an ignition source, likely a broken light tube, and ignited. The fire spread into the midsection,  where the engineer perished due to the fire. However, the fire rapidly heated the air tank of the vessel, and pressurized tank exploded, shredding 2 crew  members and critically damaged the vessel.

Incidents like this investigated by the PGFCSB clearly show a poor safety culture on the frontier.

## Why It's Good For The Game
You see that above story? This was from a real round, however the  part i neglected to mention was the massive explosion was not actually a mechanic, but an admin fucking up  blowing up an oxygen tank to try and add more oxygen for the fire, which accidently maxcaped the ship. However, I was so impressed with this story I decided to make it into a mechanic and is apart of this pr, which it's focus is workplace hazards. So let's see what we have here.

- Many changes were made to make huamans more squishy and less supermen.
-  - But wait! Won't this conflict with #3358 ? The changes  here are  made to work in tanden with that PR, for better situational awareness and to make hazards more evident.
- Ports TGMC's "flame_radius" proc and is used by flame radius for explosions
- Extingushers like flamethrowers can go over objects now, so you can finally extinguish the  fucking turf fires blocked by objects.
- Smoking a cigarette near a fuel tank or pressurized tank will either blow up the tank or set you on fire in a flash fire.
- Makes tank explosions better  and less shitty, it should be more common to happen during fires, especially during gas fires
- Adds carbon monoxide, produced by PACMANS, uranium ones produce barely any. Unlike IRL its not flammable as I cannot set the lower explosive limit for it.
- Makes hydrogen invisible. Smiles
- Pressurized tanks have much more air, totally not for them to explode more easily.
- Gibbing doesnt  destroy the body anymore. Anti-salt measure and  gibbing destroying torsos is very unpopular, might as well do it here.

## Changelog

:cl:
add: Revise your safety guidelines, as fires can and will explode tanks if in contact with them! Be careful!
add: Carbon monoxide, produced by pacmans. Don't huff it, it's not fun.
add: changes  how flame explosions work
add: Be careful where you smoke...
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/death.dm [hunk 1]
```diff
@@ -24,7 +24,7 @@
 	M.Scale(1.8, 1.2)
 	animate(src, time = 40, transform = M, easing = SINE_EASING)
 
-/mob/living/carbon/gib(no_brain, no_organs, no_bodyparts, safe_gib = FALSE)
+/mob/living/carbon/gib(no_brain, no_organs, no_bodyparts, safe_gib = TRUE)
```
Comment (FalloutFalcon) [inline code/modules/mob/living/carbon/death.dm]: This has to be a separate pr

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1185,6 +1190,92 @@
...
+		T.atmos_spawn_air("[GAS_CO]=[reac_volume/2];TEMP=[temp]")
+	return
+
+/datum/reagent/carbon_monoxide/on_mob_delete(mob/living/living_mob)
+	var/mob/living/carbon/living_carbon = living_mob
+	if(accumilation <= 140)
+		living_carbon.cure_trauma_type(/datum/brain_trauma/mild/muscle_weakness)
+		living_carbon.cure_trauma_type(/datum/brain_trauma/mild/concussion)
+		living_carbon.cure_trauma_type(/datum/brain_trauma/mild/speech_impediment)
+		living_carbon.cure_trauma_type(/datum/brain_trauma/mild/expressive_aphasia)
```
Comment (FalloutFalcon) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: you can now clear a concussion by huffing a single breath of carbon monoxide. You need to add source types if you want this so you only clear a trauma developed from carbon monoxide.

File: code/modules/atmospherics/auxgm/gas_types.dm [hunk 3]
```diff
@@ -161,7 +172,7 @@
 	specific_heat = 10
 	name = "Hydrogen"
 	flags = GAS_FLAG_DANGEROUS
-	moles_visible = MOLES_GAS_VISIBLE
+	//moles_visible = MOLES_GAS_VISIBLE
```
Comment (FalloutFalcon) [inline code/modules/atmospherics/auxgm/gas_types.dm:175]: Explain why

File: code/datums/components/pellet_cloud.dm [hunk 4]
```diff
@@ -78,6 +76,8 @@
 		RegisterSignal(parent, COMSIG_MINE_TRIGGERED, PROC_REF(create_blast_pellets))
 	else if(issupplypod(parent))
 		RegisterSignal(parent, COMSIG_SUPPLYPOD_LANDED, PROC_REF(create_blast_pellets))
+	else
+		RegisterSignal(parent, COMSIG_SUPPLYPOD_LANDED, PROC_REF(create_blast_pellets))
```
Comment (FalloutFalcon) [inline code/datums/components/pellet_cloud.dm]:
```suggestion
	else
		RegisterSignal(parent, COMSIG_SUPPLYPOD_LANDED, PROC_REF(create_blast_pellets))
```

File: code/datums/components/pellet_cloud.dm [hunk 5]
```diff
@@ -46,8 +46,6 @@
 	var/mob/living/shooter
 
 /datum/component/pellet_cloud/Initialize(projectile_type=/obj/item/shrapnel, magnitude=5)
-	if(!isammocasing(parent) && !isgrenade(parent) && !islandmine(parent) && !issupplypod(parent))
-		return COMPONENT_INCOMPATIBLE
```
Comment (FalloutFalcon) [inline code/datums/components/pellet_cloud.dm:50]: please at least check that its an atom or obj

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 6]
```diff
@@ -1185,6 +1190,92 @@
...
+		if(150 to 250)
+			to_chat(victim, "<span class='userdanger'>[pick("Your head hurts!", "You feel a burning knife inside your brain!", "A wave of pain fills your head!")]</span>")
+			victim.adjustStaminaLoss(3)
+			victim.Stun(10)
+			victim.Dizzy(5)
+			victim.confused += (accumilation/50)
+			victim.gain_trauma(/datum/brain_trauma/mild/expressive_aphasia)
+			victim.gain_trauma(/datum/brain_trauma/mild/muscle_weakness)
+		if(250 to 350)
+			to_chat(victim, "<span class='userdanger'>[pick("What were you doing...?", "Where are you...?", "What's going on...?")]</span>")
```
Comment (FalloutFalcon) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: span define all of these

File: code/modules/surgery/organs/lungs.dm [hunk 7]
```diff
@@ -323,6 +323,36 @@
...
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,2)
+				if (100 to 200)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,4)
+				if (200 to 400)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,8)
+				if (400 to INFINITY)
+					monoxide_reagent.accumilation = max(monoxide_reagent.accumilation, 450)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,16)
```
Comment (FalloutFalcon) [inline code/modules/surgery/organs/lungs.dm:347]: the way this scales could probably be done with math instead of a switch statment

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]
This is pretty unatomic.. Nearly all of this is balance changes that seem pretty arbitrary. I'm really not sure I like the tank thresholds much and would like to see some examples of how that would play out in game, are there any tanks that start with this pressure. How easy is it to actually blow up a tank.
All the current sources of gibbing SHOULD be round removal as its mostly mega fauna or extreme explosions that should not leave a intact torso. carbon monoxide also seems like it should be its own pr as i don't see its relation to explosions at all. Im also not a big fan of most of its code but i imagine all the other gasses aren't much better.
I would prefer this being broken up into at least 2 prs probably more tbh.

**Comment by generalthrax** [comment]
As of right now, the tank max pressure means that air pumps will immediately blow up tanks if you fill them.

Is carbon monoxide produced by anything other than pacmans? I've noticed that its just Everywhere on a ship even though scrubbers should be robust enough to deal with it.

File: code/modules/atmospherics/auxgm/gas_types.dm (hunk 3 above)
Comment (KittyNoodle) [reply code/modules/atmospherics/auxgm/gas_types.dm:175]: > Explain why

image

**Comment by firebudgy** [comment]
Ships aren't properly scrubbing CO in engine bays, and apparently it makes you permanently hallucinate upon entering them. Might be something worth tuning.

**Comment by Gristlebee** [comment]
I'd maybe recommend adding some hint or addition to the descs welding tanks and gas tanks to indicate it can explode if you smoke near them, so there's some stated warning to a punishment mechanic.

**Comment by KittyNoodle** [comment]
a major problem with Carbon Monoxide is that it works like BZ, so if you take in a whif of a tiny amount of the gas you just get fucked for the next 7 minutes

**Comment by firebudgy** [comment]
Probably worth adding some form of indication to exploding emergency oxygen tanks and the like since nobody seems to know what the safe pressure is (as there's no feedback).

File: code/__DEFINES/atmospherics.dm [hunk 8]
```diff
@@ -168,13 +168,13 @@
...
 //TANKS
 /// temperature in kelvins at which a tank will start to melt
-#define TANK_MELT_TEMPERATURE 1000000
+#define TANK_MELT_TEMPERATURE 1000 + T0C
 /// Tank starts leaking
-#define TANK_LEAK_PRESSURE (30.*ONE_ATMOSPHERE)
+#define TANK_LEAK_PRESSURE (20 * ONE_ATMOSPHERE + 5)
 /// Tank spills all contents into atmosphere
-#define TANK_RUPTURE_PRESSURE (35.*ONE_ATMOSPHERE)
+#define TANK_RUPTURE_PRESSURE (20 * ONE_ATMOSPHERE)
```
Comment (SomeguyManperson) [inline code/__DEFINES/atmospherics.dm:175]: 2023 kpa is less than the air pump, which fills to ~2533, so people trying to fill tanks with one can break them (source: experience). I'd think at the very least tanks should have warning labels for maximum pressure/temperature

File: code/game/objects/items/tanks/tanks.dm [hunk 9]
```diff
@@ -219,8 +219,37 @@
...
+	var/tank_temperature = air_contents.return_temperature()
+	tank_temperature += exposed_temperature/20 //slowly equalize with the air, since this is over an active fire, we heat up faster
+	air_contents.set_temperature(tank_temperature)
+	if(exposed_temperature > TANK_MELT_TEMPERATURE)
+		take_damage(max((exposed_temperature - TANK_MELT_TEMPERATURE)/2, 0), BURN, 0)
+	if(exposed_volume > TANK_RUPTURE_PRESSURE) // implosion
+		take_damage(max((exposed_volume - TANK_RUPTURE_PRESSURE)/2, 0), BURN, 0)
+
+
+/obj/item/tank/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
```
Comment (SomeguyManperson) [inline code/game/objects/items/tanks/tanks.dm:238]: it's easy for a tank to end up overheating on lava planets & there's no fast way to un-overheat a tank as far as I'm aware

File: code/modules/power/port_gen.dm [hunk 10]
```diff
@@ -94,6 +94,8 @@
 	var/sheet_left = 0 // How much is left of the sheet
 	var/time_per_sheet = 260
 	var/current_heat = 0
+	var/pollution_multiplier = 0
+	var/pollution_gas = GAS_CO
```
Comment (SomeguyManperson) [inline code/modules/power/port_gen.dm:98]: This stuff feels like it would work better if not for the fact you are inside a ship all the time, moving a generator outside would be complicated and/or tedious and most ships rely on them

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 11]
```diff
@@ -1185,6 +1190,89 @@
...
+	description = "A highly dangerous gas for sapients."
+	reagent_state = GAS
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	color = "#96898c"
+	var/accumilation
+
+/datum/reagent/carbon_monoxide/on_mob_life(mob/living/carbon/victim)
+	if(holder.has_reagent(/datum/reagent/oxygen))
+		holder.remove_reagent(/datum/reagent/carbon_monoxide, 2*REM)
+		accumilation = accumilation/4
```
Comment (SomeguyManperson) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1204]: This is double dipped w/ oxygen reagent on_mob_life and I think it should be handled in one or the other

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 12]
```diff
@@ -1185,6 +1190,89 @@
...
+		if(50 to 150)
+			to_chat(victim, "<span class='warning'>[pick("Your head hurts.", "Your head pounds.")]</span>")
+			victim.Dizzy(5)
+		if(150 to 250)
+			to_chat(victim, "<span class='userdanger'>[pick("Your head hurts!", "You feel a burning knife inside your brain!", "A wave of pain fills your head!")]</span>")
+			victim.Stun(10)
+			victim.Dizzy(5)
+			victim.confused = (accumilation/50)
+			victim.gain_trauma(/datum/brain_trauma/mild/expressive_aphasia)
+			victim.gain_trauma(/datum/brain_trauma/mild/muscle_weakness)
```
Comment (SomeguyManperson) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1219]: The traumas here should probably be tracked somehow so huffing monoxide doesn't fix unrelated ones

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 13]
```diff
@@ -1185,6 +1190,89 @@
 		M.confused = min(M.confused + 2, 5)
 	..()
 
+/datum/reagent/carbon_monoxide
+	name = "Carbon Monoxide"
+	description = "A highly dangerous gas for sapients."
+	reagent_state = GAS
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
```
Comment (SomeguyManperson) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: 0.04 metabolization speed means it'll likely need to be purged externally if you breathe any, and while injecting reagent oxygen might be intuitive for people who know how it works its probably not intuitive for a player

File: code/__DEFINES/atmospherics.dm [hunk 14]
```diff
@@ -168,13 +168,13 @@
 
 //TANKS
 /// temperature in kelvins at which a tank will start to melt
-#define TANK_MELT_TEMPERATURE 1000000
+#define TANK_MELT_TEMPERATURE 1000 + T0C
 /// Tank starts leaking
-#define TANK_LEAK_PRESSURE (30.*ONE_ATMOSPHERE)
+#define TANK_LEAK_PRESSURE (20 * ONE_ATMOSPHERE + 5)
```
Comment (SomeguyManperson) [inline code/__DEFINES/atmospherics.dm:173]: This is higher than the rupture pressure so tanks probably won't leak (for long) they'll just pop

File: code/__DEFINES/species.dm [hunk 15]
```diff
@@ -1,12 +1,12 @@
...
+#define HAZARD_HIGH_PRESSURE 303
 /// This determins when the orange pressure icon is displayed (it is 0.7 * HAZARD_HIGH_PRESSURE)
-#define WARNING_HIGH_PRESSURE 325
+#define WARNING_HIGH_PRESSURE 202
 /// This is when the gray low pressure icon is displayed. (it is 2.5 * HAZARD_LOW_PRESSURE)
-#define WARNING_LOW_PRESSURE 50
+#define WARNING_LOW_PRESSURE 60
 /// This is when the black ultra-low pressure icon is displayed. (This one is set as a constant)
-#define HAZARD_LOW_PRESSURE 20
+#define HAZARD_LOW_PRESSURE 40
```
Comment (SomeguyManperson) [inline code/__DEFINES/species.dm]: probably update the comments here because they will be inaccurate with the current numbers

File: code/__DEFINES/species.dm [hunk 16]
```diff
@@ -28,11 +28,11 @@
 /// The body temperature limit the human body can take before it starts taking damage from heat.
 /// This also affects how fast the body normalises it's temperature when hot.
 /// 340k is about 66c, and rather high for a human.
-#define HUMAN_BODYTEMP_HEAT_DAMAGE_LIMIT (HUMAN_BODYTEMP_NORMAL + 30)
+#define HUMAN_BODYTEMP_HEAT_DAMAGE_LIMIT (HUMAN_BODYTEMP_NORMAL + 20)
 /// The body temperature limit the human body can take before it starts taking damage from cold.
 /// This also affects how fast the body normalises it's temperature when cold.
 /// 270k is about -3c, that is below freezing and would hurt over time.
-#define HUMAN_BODYTEMP_COLD_DAMAGE_LIMIT (HUMAN_BODYTEMP_NORMAL - 40)
+#define HUMAN_BODYTEMP_COLD_DAMAGE_LIMIT (HUMAN_BODYTEMP_NORMAL - 30)
```
Comment (SomeguyManperson) [inline code/__DEFINES/species.dm]: Update the comments here too, new numbers are 57 c/ 330 k and 7 c/ 280 k

File: code/game/objects/items.dm [hunk 17]
```diff
@@ -832,6 +832,29 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 
+	if(!prob(1))
+		return
+
+	for(var/dir in GLOB.cardinals)
+		var/ruined_round = FALSE
+		var/turf/test_turf = get_step(location, dir)
+		for(var/obj/to_test as obj in test_turf.contents)
+			if(istype(to_test, /obj/structure/reagent_dispensers/fueltank))
+				location.visible_message("<span class='userdanger'>A single ember from [src] drops gently onto [to_test]. Uh oh.</span>")
```
Comment (SomeguyManperson) [inline code/game/objects/items.dm:843]: span macros

File: code/game/objects/items/tanks/tanks.dm (hunk 9 above)
Comment (rye-rice) [reply code/game/objects/items/tanks/tanks.dm:238]: I removed this mechanic, temperature_expose only procs on fires

Before I removed it putting the tank back in your ship would cool it ambiently

File: code/__DEFINES/atmospherics.dm (hunk 8 above)
Comment (rye-rice) [reply code/__DEFINES/atmospherics.dm:175]: Good idea

Probably on a examine()

File: code/modules/power/port_gen.dm (hunk 10 above)
Comment (rye-rice) [reply code/modules/power/port_gen.dm:98]: Yeah it's effectively disabled right now, atmos during testing and on live looked very different. I might keep it then enable it far far later when we got better primary engines

Comment (generalthrax) [reply code/modules/power/port_gen.dm:98]: iirc Rylies ripping this out for now, for a separate pr.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 12 above)
Comment (rye-rice) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1219]: There's no way to track it as far as I'm aware, the < 140 is a bandaid

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 13 above)
Comment (rye-rice) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Yeah
Probably going to make it a normal rate, this was from when I couldn't get it to disappear immediately

File: code/__DEFINES/atmospherics.dm (hunk 14 above)
Comment (rye-rice) [reply code/__DEFINES/atmospherics.dm:173]: I removed tank leaking from internal tanks, it didn't seem to even work as intended and during testing it was like... meh honestly

File: code/__DEFINES/species.dm (hunk 15 above)
Comment (rye-rice) [reply code/__DEFINES/species.dm]: Yep

File: code/__DEFINES/species.dm (hunk 16 above)
Comment (rye-rice) [reply code/__DEFINES/species.dm]: Ty

**Comment by rye-rice** [comment]
I'll try later to get this into a mergable state but tmt wanted to look at it post pref rework, so don't merge until then

File: code/modules/mob/living/carbon/death.dm (hunk 1 above)
Comment (thgvr) [inline code/modules/mob/living/carbon/death.dm]: Again, please stop shoving this into random PRs without talking to anyone about it. I'm not interested in having extreme explosions just blow your limbs off like a lego man

Comment (FalloutFalcon) [reply code/modules/mob/living/carbon/death.dm]: additionally if your going to make the change because you want to prevent it specifically for explosion related gibbing it needs to be for those only instead of turning most if not all gibs into safe ones.
make this its own pr please..... and actually elaborate why..

**Comment by thgvr** [comment]
I'm almost sure now that setting `safe_gib = TRUE` will crash the server on gibbing

**Comment by rye-rice** [comment]
removed safe_gib from this, i still think its a good idea but it crashes the server unfortunately, for another time

im moving the temperture stuff to another pr

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 13 above)
Comment (rye-rice) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: changed

File: code/__DEFINES/species.dm (hunk 16 above)
Comment (rye-rice) [reply code/__DEFINES/species.dm]: moved to another pr

**Comment by FalloutFalcon** [comment]
> removed safe_gib from this, i still think its a good idea but it crashes the server unfortunately, for another time
> 
> im moving the temperture stuff to another pr

this should have been removed from the pr regardless of if it crashed.

**Comment by rye-rice** [comment]
For another time
